homework-submission/reactweek1/src/App.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import './App.css';
+import list from './list.json';
+class App extends Component {
+  constructor(props) { 
+    super(props);
+    this.changeToDone = this.changeToDone.bind(this);
+    this.state = {
+      list
+    }
+  }
+  changeToDone = (event) => {
+    const id = event.target.id;
+    const listCopy = this.state.list.slice(0);
+    const itemIndex = listCopy.findIndex(el => el.id == id);
+    listCopy[itemIndex].done = !listCopy[itemIndex].done;
```

Comment:
Thanks for reply @amirno1, Yes this is why this whole review on week2 was all about state.
it's just to make students catch up because state is a very main thing in react, that whole libraries built on top of it.

And I think you asked a very good question here and the answer want some how to expand the explain further more, do you have some time on Thursday 9 Pm, so we make sure we can follow the road and catch up on every thing?

Reply:
Yes it's via slack.